[WIP] Introduce logging extensibility, JSON formatter extension

dmlloyd opened the pull request:
Fixes #1847, fixes #2409, enables #3985, enables #3166, enables #453, replaces #2348

Sanne committed:
Upgrade to Hibernate ORM 5.4.9.Final

dmlloyd committed:
Add pluggable log handler and formatter capability

Co-authored-by: ia3andy <adamevin@redhat.com>

dmlloyd committed:
Introduce JSON formatting extension

stuartwdouglas reviewed (dismissed):
The implementation looks good to me, but there are no docs or tests. This should have an entry in the logging guide, and ideally some kind of testing as well

ia3andy commented on core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java:
```diff
@@ -94,6 +99,14 @@ public void initializeLogging(LogConfig config) {
             configureSyslogHandler(config.syslog, errorManager, filterElements, handlers);
         }
 
+        final ErrorManager finalErrorManager = errorManager;
```
Why is having this `finalErrorManager` necessary?

dmlloyd replied:
Because it is captured in a lambda.  Or I mean... it used to be... oops :)

dmlloyd replied:
Fixed.

ia3andy commented:
@dmlloyd thanks for making me co-author :)

I know it's not related to the PR but this method would factor the same code used 3 times:
```
 private static AsyncHandler createAsyncHandler(AsyncConfig asyncConfig, Level level, Handler handler) {
        final AsyncHandler asyncHandler = new AsyncHandler(asyncConfig.queueLength);
        asyncHandler.setOverflowAction(asyncConfig.overflow);
        asyncHandler.addHandler(handler);
        asyncHandler.setLevel(level);
        return asyncHandler;
    }
```

```
if (config.async.enable) {
                handlers.add(createAsyncHandler(config.async, config.level, handler));
```

ia3andy committed:
Slightly clean LoggingSetupRecorder for readability

ia3andy commented:
@dmlloyd if you don't like my commit, feel free to edit/remove it ;-)

I just thought that a few things where hard to read like list mutated by methods..

geoand commented:
@ia3andy since you have made changes, would you mind looking into creating a test and some documentation if @dmlloyd isn't already on it?

dmlloyd commented:
> @dmlloyd if you don't like my commit, feel free to edit/remove it ;-)
> 
> I just thought that a few things where hard to read like list mutated by methods..

Yeah there are things that can be improved in the original code.  I think that might be best done as another PR or separate commits at least.

dmlloyd committed:
Add documentation for the JSON logging extension

gastaldi commented on docs/src/main/asciidoc/logging-guide.adoc:
```diff
@@ -133,6 +133,58 @@ The logging format string supports the following symbols:
+  <dependencies>
+    <!-- ... your other dependencies are here ... -->
+    <dependency>
+      <artifactId>quarkus-logging-json</artifactId>
```
no `<groupId>` ?

dmlloyd replied:
Fixed.

ia3andy commented:
@dmlloyd I've started creating test, I have a problem with my JDK to run them. I'll continue on monday. Have a great WE cheers

ia3andy committed:
Add tests for json formatter and custom log handlers

michael-simons committed:
Upgrade to Neo4j driver 4.0.0-beta.

The next version will include substitutions and neccessary instructions to make our driver work natively, so those substitutions are no longer necessary to have in quarkus.

wburns committed:
Update Infinispan to 10.0.0.Final

ia3andy commented:
@dmlloyd I got stuck with the baby 😂I'll get back to this one during the week..

dmlloyd commented:
No problem, I've been stuck with some things too.  Looks like there are some conflicts to iron out too now (I can take care of that).

CSTDev committed:
Add option to configure the engine to require/request client authentication.

CSTDev committed:
Enable setting of truststore path, type and password

ia3andy commented:
I created an issue for testing:
- @dmlloyd could u rebase?
- @gsmet I think after rebase it's ready to be merged

dmlloyd commented:
Again! sure, no problem :)

ochaloup committed:
[#4765] fix node name to be harmonized with xa node name

CSTDev committed:
Add test for Mutual TLS

Also allow the SSL URL to be injected into tests that need it.

loicmathieu committed:
doc: how to access the config programmatically

CSTDev committed:
Add jks as resource for native tests

iocanel committed:
feat: Add support for generating knative manfiests.

iocanel committed:
test: Add integration test that tests the generation of knative manifests.

iocanel committed:
doc: How to generate and customize knative manifests.

iocanel committed:
chore: bump dekorate to 0.9.9.

CSTDev committed:
Use correct ports for MTLS native test

Native runs in prod profile and so doesn't use the set test ports,
need to explicitly use the same port as would be used in prod. Currently hardcoded though.

Use different method to include resources, specify all jks rather than using resource file.

Fix import order

Correct algorithm used in keystores, can't use DSA in Java 11+

mmusgrov committed:
issue 4791 Do not open ports at build time (disable TSM)

oztimpower committed:
Kotlin requires all-open plugin

sberyozkin committed:
Updating to quarkus-vertx-core and other dep optimizations

loicmathieu commented:
I need JSON logging because of the unability to a lot of log tools to correclty handle multi-line logs ...
Is there a chance to see this merged soon ?

Sanne committed:
Complete the Hibernate ORM guide, clarify the limitations

ia3andy commented:
@loicmathieu it was supposed to be in 0.27.0, I guess we missed it in the rush.. @gsmet I'll rebase, can you mark it for the next one?

michalszynkiewicz committed:
vertx idle timeout exposed in the http configuration

refs #4455

michalszynkiewicz committed:
CDI security determined at build time, fixes #4992

geoand committed:
Provide the ability to remove final flag from methods of CDI beans

We need security annotations to result in the creation of an interceptor
which is not possible when a method is final.
The solution we follow is to remove the final modifier from methods
that need intercepting.
This is now configurable with the default value true, meaning that
Arc will remove the final flag. If the value is set to false
Arc throws an exception at build time.

Fixes: #5051

rsvoboda committed:
Removal of unused appModel

mkouba committed:
Docs - CDI guide - introduce the "Non-standard Features" section

- also try to clarify "Native Executables and Private Members" section

rsvoboda committed:
Use %n for new line in case of String.format

mkouba committed:
Update docs/src/main/asciidoc/cdi-reference.adoc

Co-Authored-By: Matej Novotny <matej.novotny2@gmail.com>

mkouba committed:
Update docs/src/main/asciidoc/cdi-reference.adoc

Co-Authored-By: Matej Novotny <matej.novotny2@gmail.com>

mkouba committed:
Update docs/src/main/asciidoc/cdi-reference.adoc

Co-Authored-By: Matej Novotny <matej.novotny2@gmail.com>

mkouba committed:
Update docs/src/main/asciidoc/cdi-reference.adoc

Co-Authored-By: Guillaume Smet <guillaume.smet@gmail.com>

aloubyansky committed:
Use ThreadLocal to store the global value only for gradle add and list extensions tasks, otherwise the default global config should use a static field

gwenneg committed:
Fix conditions that always evaluate to false

gwenneg committed:
Fix potential NPEs

yrodiere committed:
Move mapper configuration out of the "elasticsearch" section in the Hibernate Search extension config

For consistency, because we already do this for:

* default-backend
* background-failure-handler

So we may as well do it for these, too:

* query-loading.cache-lookup.strategy
* query-loading.fetch-size
* automatic-indexing.synchronization.strategy
* automatic-indexing.enable-dirty-check

yrodiere committed:
Move configuration of the Elasticsearch backend to a dedicated section in the Hibernate Search extension docs

yrodiere committed:
Move the "default backend" configuration to the "additional backends" section in the Hibernate Search extension config

Because having it at the very top of the property list in the
documentation is quite confusing, considering it's really not something
the average user will need.

yrodiere committed:
Rename "elasticearch" to "defaultBackend" in the build time Hibernate Search configuration

For consistency with the runtime configuration.

Sanne committed:
Make VertxHttpRequest a final class

Sanne committed:
Performance: VertxHttpRequest should allocate attributes map lazily

Sanne committed:
Performance: try resolve the remote Host address lazily

Sanne committed:
Remove some dead code

gastaldi committed:
Replace gradle-tooling-api with gradleApi()

As reported in #5204

machi1990 committed:
fix: Dev mode - 404 Page - Live reloaded list of static files

FroMage committed:
Fix for #5214

Replace $ with . in loaded class names

FroMage committed:
Fix all-config odd/even rows by not looking into member tables

FroMage committed:
Workaround #5214 for openapi table

gsmet commented:
Let's try to target 0.29 for that one. @ia3andy  will work on some tests next week.

gsmet committed:
Fix the Infinispan Embedded guide: it breaks the website build

gastaldi committed:
Gradle addExtension now handles comma-separated values

Fixes #5224

machi1990 committed:
feat: list openapi, swaggerui, metrics and health endpoints in not found page

Fixes https://github.com/quarkusio/quarkus/issues/5150

machi1990 committed:
fix: make sure that cachedOpenApiPath is always resolved against http rootpath

maxandersen committed:
Images through imageoptiom

shaved 0.63 MB of the images.

$ git file-size-diff --cached master
-31745    docs/src/main/asciidoc/images/amqp-guide-architecture.png
-9216    docs/src/main/asciidoc/images/architecture-phases.png
-23195    docs/src/main/asciidoc/images/containerization-process.png
-12213    docs/src/main/asciidoc/images/getting-started-architecture.png
-72545    docs/src/main/asciidoc/images/http-architecture.png
-31325    docs/src/main/asciidoc/images/kafka-guide-architecture.png
-55993    docs/src/main/asciidoc/images/kafka-streams-guide-architecture-distributed.png
-58147    docs/src/main/asciidoc/images/kafka-streams-guide-architecture.png
-12353    docs/src/main/asciidoc/images/kogito-guide-screenshot.png
-19592    docs/src/main/asciidoc/images/native-executable-process.png
-53760    docs/src/main/asciidoc/images/openapi-swaggerui-guide-screenshot01.png
-161907    docs/src/main/asciidoc/images/openapi-swaggerui-guide-screenshot02.png
-14348    docs/src/main/asciidoc/images/scheduling-task-architecture.png
-15774    docs/src/main/asciidoc/images/validation-guide-architecture.png
-14666    docs/src/main/asciidoc/images/validation-guide-screenshot.png
-16238    docs/src/main/asciidoc/images/websocket-guide-architecture.png
-22729    docs/src/main/asciidoc/images/websocket-guide-screenshot.png
total -625746

gastaldi committed:
Mailer: Handles Reply-To correctly

Fixes #5231

aguibert committed:
Fix incorrect GraalVM option and use correct build path for Gradle resource config file

aguibert committed:
#5138 Drop 'classpath:' protocol prefix for Flyway migration locations

stuartwdouglas committed:
Force SecurityCheckStorage to be an application class

jaikiran committed:
Fix Kafka test to handle empty consumer records on poll

jaikiran committed:
Use unique name for the producer, created in the Kafka streams test

machi1990 committed:
refactor(vertx-graphql): remove unused CombinedIndexBuildItem parameter and move the feature label constants to FeatureBuildItem labels blocks

jaikiran committed:
issue-5247 Work around an issue in vertx where getting remote/client SocketAddress of the request leads to a NPE

gsmet committed:
Switch all the cross-guides links to use the canonical website url

jorsol committed:
chore(deps): update okhttp to 3.12.6

aloubyansky committed:
Make the create project mojo resolve the latest version of the universe by default and fall back to the bundled quarkus-bom in offline mode

Sanne committed:
Trivial: fix a link anchor name

sberyozkin committed:
Update to smallrye-jwt-2.0.10

rsvoboda committed:
Unused imports removal

stuartwdouglas committed:
Remove augmentation messages from test and dev mode

stuartwdouglas committed:
Fix CI

mkouba committed:
Support repeatable QuarkusTestResource annotation

- resolves #5256

Sanne committed:
Remove the lambda use in RESTEasy's VertxRequestHandler

manovotn committed:
Arc context propagation - revert to sharing who concurrent map, add automated test.

marcinczeczko committed:
Fix for #5115 - Do not fire recorder if no DynamoDB clients used

mswiderski committed:
Updated Kogito extension description and labels

johnoliver committed:
Add docs for fixing read only file system deployments

Commit:
fix incorrect delegation usage in ResumingRouter

sean-scott-lr commented:
Since this was targeting 0.29.0 will this be put into 1.0 instead?

sberyozkin committed:
Link RT with AT, moving ID and AT token credentials out of 'runtime' subpackage

aoudiamoncef committed:
style(*): Propagate Jboss Logmanager

Commit:
hibernate validator: Remember annotated interface methods
resolves #1888

FroMage committed:
Fix for 5274 (#5276)

* Panache: do not create findById bridge for abstract entity repositories

If the entity type is not fixed in the hierarchy yet, don't generate the bridge

Fix #5274

* Test for #5274: duplicate findById for abstract repositories

* Fix #5274 for Mongo too

* Test for #5274 for Mongo

wburns committed:
Fixed issue where app class couldn't be loaded in dev mode

stuartwdouglas commented on docs/src/main/asciidoc/logging.adoc:
```diff
@@ -75,6 +75,59 @@ The logging format string supports the following symbols:
+In order to configure JSON logging, the `quarkus-logging-json` extension may be employed.  Add this extension to your
+application POM as the following snippet illustrates.
+
+NOTE: This snippet assumes that your application POM is configured in the manner described in the
```
This note just trails off.

stuartwdouglas reviewed (changes requested):
Other than the minor docs issue I think it is good to go

gsmet commented:
@stuartwdouglas @ia3andy is planning to work on tests. I would prefer we have tests before merging.

aguibert committed:
Update Kotlin and Scala gradle templates to match Java template

gsmet committed:
Remove the README.md in integration-tests

It's totally outdated and doesn't have any useful information.

Let's delete it, I think we have enough information elsewhere.

Fixes #1289

stuartwdouglas committed:
Provide a way to set the TCCL in @BuildStep

This allows for the TCCL to automatically set the TCCL
in a build step. We can't do it for everyone as the metadata
needed to build this CL is created by @BuildStep methods.

Any build step that needs to load application classes should
use this flag to get a transformation safe ClassLoader.

stuartwdouglas committed:
Fix OIDC flow issue

This would attempt to authenticate for unprotected resources

jaikiran committed:
issue-5293 Fix NPE when processing a @WebServlet which has additional annotations other than @ServletSecurity

loicmathieu commented:
@dmlloyd  I start prototyping an extension to be able to send log events to Graylog or Logstash based on the log customization infrastructure you implement in this PR.

It works great, but it raises the question of naming log related extension, and you know, naming is one of the most complex task in programming ;)

You create an extension named `logging-json` that defines a log formatter for JSON output.

My extension is a log handler for GELF backend (Logstash or gGraylog), For the moment I named it `gelf-log-handler`.

If we develop several log extensions, better have them all in the same namespace, so I should rename mine `logging-gelf` or `logging-handler-gelf. If we choose the latest your extension should be renamed `logging-formatter-json` to be consistent I think ...

marcinczeczko committed:
Fix for #5286 - Split config into build & runtime configs. Move runtime config validation to processor

geoand committed:
Ensure that array config values are properly validated at startup

Fixes: #5297

anamarija committed:
protectionDomain creation for byteContent class definition for RuntimeClassLoader

jmartisk committed:
[#5318] Properly match paths ending with /*

machi1990 committed:
fix: remove unsupported native-image flags

this remove the enableRetainedHeapReporting i.e -H:+PrintRetainedHeapHistogram and
enableCodeSizeReporting i.e -H:+PrintCodeSizeReport

fixes #5324

edewit committed:
fixed typo

sberyozkin committed:
Restore one of RT classes, produce all token credentials

wburns committed:
Make Infinispan embedded cache manager a singleton scoped to an application

galderz committed:
Use separate URLClassPath targets depending on the JDK version #5335

Sanne committed:
Even when Hibernate ORM is not being used, the feature static initializer must be invoked

Sanne committed:
Move required System Properties to standard Build Items

dmlloyd commented:
I think `logging-gelf` makes sense.  I think it's unlikely that we'll ever need to disambiguate between handlers and formatters - if some technology covers both, I'd expect we'd want to ship a single extension that did both.

Sanne committed:
Add a new integration test to verify native compilation with JPA feature disabled

machi1990 committed:
refactor: uses UncheckedIOException instead of RuntimeException

jaikiran committed:
Include the original exception that caused classloading to fail

rsvoboda committed:
ZipResourceLoader fix to compile with Java 13

geoand committed:
Ensure that the name of the generated Spring data repositories is unique

Fixes #5317

machi1990 committed:
ci: use the recommended way to build native image in CI

gastaldi committed:
Add ADOPTERS.md

gwenneg committed:
Clean code after static code analysis

machi1990 committed:
refactor: remove unused portProducer parameter and use runtime httpConfiguration in runtime recording

jorsol committed:
chore(deps): update kubernetes-client to 4.6.3

jorsol committed:
chore(deps): update postgresql-jdbc to 42.2.8

machi1990 committed:
fix: use unique key name when generating config section anchor

The issue was reported here - https://github.com/quarkusio/quarkus/pull/4529#issuecomment-551883480

gastaldi committed:
Generated Gradle code should use the build/ directory

Fixes #5363

jaikiran committed:
Log the dev mode JVM startup command at INFO level

iocanel committed:
feat (#5345): Introduce KubernetesRoleBuildItem to allow extensions
express RBAC requirements.

iocanel committed:
feat (#5345): kubernetes-client extension now creates a view KubernetesRoleBuildItem.

iocanel committed:
test (#5345): Add integration test that verifies that the view role is
added to generated manifests when the `kubernetes-client` extension is added.

schulzp commented:
Hi thanks your work so far, I'm looking forward to this feature. @dmlloyd is @stuartwdouglas' [comment](https://github.com/quarkusio/quarkus/pull/4121/files/4326b8faa059a3b1626a0969a4818a564070696d#r343900094) the only thing blocking this PR?

machi1990 committed:
ci: activate CI for jpa-without-entity integrations native test

This is based on https://github.com/quarkusio/quarkus/pull/5343#issuecomment-552371739

iocanel committed:
doc (#5345): doucment how kubernetes extension deals with RBAC when
kubernetes-client is also present

gastaldi committed:
Remove unused gradle build parameters

oztimpower committed:
Add SAM CLI local test detail to the guide

jaikiran committed:
issue-5326 Add a testcase to reproduce the issue

gastaldi committed:
Remove unused methods in QuarkusExtension

Commit:
Add a test for a subclass method

dmlloyd commented:
Yeah as far as I know it just needs some tests.

Commit:
editorial clean up

stuartwdouglas committed:
Load classes using Class.forName

We need to do this as the class may be package private.

Fixes #5326

stuartwdouglas committed:
Fix CI timeout by splitting up native job

stuartwdouglas committed:
Fix GraphQL dependencies

hantsy committed:
Add exception handler description in Spring web docs

ia3andy commented:
@schulzp @dmlloyd, as commented above, I am on it. Monday was a holiday in France.

mkouba committed:
Interceptor resolution - fix handling of default values of bindings

- resolves #5373

gsmet commented:
For the record, this won’t be included in 1.0.0.Final so we have time to include it properly. No need to rush. 

> Le 12 nov. 2019 à 09:08, Andy Damevin <notifications@github.com> a écrit :
> 
> @schulzp @dmlloyd, as commented above, I am on it. Monday was a holiday in France.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

geoand committed:
Make @ResponseStatus on a void controller advice method optional

Fixes: #5392

michalszynkiewicz committed:
CDI Security: loop through index only if necessary

machi1990 committed:
refactor(spring-web): remove validate method after actual validation were removed

The actual validations were removed in 6a6d989d190c6cde080da72e9e37c1d4523e9a2e and
bad1edcd2a2e0bc8586f38cab664b19d646987e3

gastaldi committed:
Remove unused Main-Class parameter from Gradle plugin

sebastienblanc committed:
removing reference to the authorization part

geoand committed:
Ensure that the manifest of the generated jar can be read using JarInputStream

Fixes: #5399

Co-authored-by: Jaikiran Pai <jaikiran.pai@gmail.com>

kenfinnigan committed:
Fix failing tests in master

maxandersen committed:
enable stalebot

sberyozkin committed:
Fix the application-type property in the oidc web authentication guide

geoand committed:
Take Spring Data Pageable's Sort into account

Fixes #5406

cescoffier committed:
Avoid the re-creation of the event loops

It was missing a @BuildItem.

aoudiamoncef committed:
build(MongoDB): Remove transitive dependencies

dmlloyd committed:
Disable sporadically failing dev mode test

See #5424

akochnev committed:
Updated mailer docs with option for google password auth , issue #4510

By default, the vertx mail client and google both default to XOAUTH2 and that method needs to be disabled in order for password authentication (with app passwords to work)

Addresses issue https://github.com/quarkusio/quarkus/issues/4510

stuartwdouglas committed:
Fix expression to disable expansion pool

geoand committed:
Prevent @Bean annotated methods from producing a bean if class doesn't use @Configuration

Fixes: #5393

maxandersen committed:
more informative text for stale_bot + diff limits for PR and issues

geoand committed:
Add conversion table from Spring DI annotations to CDI / MicroProfile

geoand committed:
Add conversion table from Spring Web annotations to JAX-RS

tiagodolphine committed:
Fix Issue 5440 - Link to Infinispan Query DSL is broken on the documentation

maxandersen committed:
remove stalebot for now

gastaldi committed:
Add Vodafone Greece to ADOPTERS.md

Commit:
Use same ResumingRouter in CDI as in Observes events.

pedroigor committed:
[fixes #5323] - Only SERVICE type applications are supported

mkouba committed:
Consolidate scheduler extension - part 1

- introduce lightweight scheduler implementation in quarkus-scheduler
- introduce quarkus-quartz based on quarkus-scheduler build steps and API
- remove Scheduler#startTimer() method
- add proper intergration test

stuartwdouglas committed:
Change the container build detection for Kafka Streams

michalszynkiewicz committed:
CDI Security: don't loop through index

FroMage committed:
Fix for #1942

Link the rest client/server guides together

mkouba committed:
ArC - fix resolution if the required type is wildcard

- resolves #3311

geoand committed:
Make build registration of security interceptors more flexible

This is done by allowing the registration of arbitrary
SecurityCheck implementations.
Furthermore the use of the AnnotationStore is hidden so as
to not give the impression that it's a feature that magically
makes security annotations work

This change paves the way for #5225 and more related improvements
in the future

gsmet committed:
Fix various typos recently introduced in the Transaction guide

See https://github.com/quarkusio/quarkus/pull/4806 .

gsmet committed:
Fix docs module description in pom.xml

machi1990 committed:
docs: change servlet to route in swagger-ui path documentation since the endpoint is not served by servlets anymore

emmanuelbernard committed:
issue 5463 - Clarify Hibernate ORM property setting in prod

gastaldi committed:
Add oidc to OpenID Connect keywords

Fixes #5472

Sanne committed:
Remove duplicate dependency from integration test module

FroMage committed:
Fix #2299: mention service files in extension guide

asoldano committed:
Upgrade to RESTEasy 4.4.1.Final

gsmet committed:
Improve formatting and use Gmail instead of google

pedroigor committed:
[fixes #5397] - Updating guide to include path permissions

gastaldi committed:
Show process exit code on error

Commit:
Spring data jpa with MappedSuperclass, fixes #5261

johnaohara committed:
Register a close handler and test for closed client connections

gwenneg committed:
Introduce UnsafeAccessedFieldBuildItem

aloubyansky committed:
If the local Maven repo includes the metadata for the default platform artifacts but the artifacts themselves are missing, the platform resolver running in offline mode should still fallback to the plaform descriptor bundled with the tools unless the user has explicitly requested the desired platform coordinates

stuartwdouglas committed:
Gizmo 1.0.0.Final

stuartwdouglas committed:
Quarkus Security 1.0.0.Final

stuartwdouglas committed:
Quarkus HTTP 3.0.0.Final

CSTDev committed:
Work around SunEC warning

maxandersen committed:
Bootstrap MavenArtifactResolver should not use the default user local repo as a fallback one for the failed artifact resolving requests in case it was setup with a custom local repo path. A new config option reTryFailedResolutionsAgainstDefaultLocalRepo was introduced on its builder to support the fallback behavior.

geoand committed:
Show more robust gradle graal configuration in docs

geoand committed:
Add more tests for ResponseEntity handling

Relates to: #5374

xfh committed:
add hibernate-jpamodelgen to bom

To generate the static metamodel, e.g. for criteria queries, the
hibernate-jpamodelgen dependency must be added to the classpath.

Adding the dependency in the dependency management section of the
bom makes it easier to keep the versions of hibernate and the
metamodel generator in sync.

manovotn committed:
Arc - support static producer methods.

aloubyansky committed:
If quarkus-platform-descriptor-json:jar couldn't be resolved and the user didn't provide any specific platform coordinates then fallback to the platform bundled with the tools

cescoffier committed:
Explicitly disable the clustering on the web Vert.x instance.

Fix #5240.

Note that again, clustering is only supported in JVM mode and requires a cluster manager in the classpath.

machi1990 committed:
refactor: remove trailing usage of @SubstrateTest

follows up https://github.com/quarkusio/quarkus/pull/5007

aloubyansky committed:
Replacing quarkus.properties in projection generation templates with quarkus.platform.version and quarkus-plugin.version for Maven and quarkusVersion with quarkusPluginVersion for Gradle, also adding quarkus.platform.group-is and quarkus.platform.artifact-id to pom.xml templates and sorting of properties before persisting pom.xml.

aoudiamoncef committed:
build(BOM): Upgrade Swagger UI to 3.24.3

machi1990 committed:
fix(logging): close handlers and free all associated resources.

I noticed that the handlers were never closed while trying to figure out the causes for
https://github.com/quarkusio/quarkus/issues/5522.

Fixes https://github.com/quarkusio/quarkus/issues/5522

machi1990 committed:
refactor: rename META_INF_SERVICES to META_INF_RESOURCES in ResteasyHotReplacementSetup

I noticed the typo when I was going around trying to fix something. While I was there, I did some
all around clean ups of variable naming

machi1990 committed:
fix: remove build warning by making injected spring data jpa fragments protected in a generated repository

This was causing the following warning:

[INFO] [io.quarkus.arc.processor.BeanProcessor] Found unrecommended usage of private members (use package-private instead) in application beans:
- @Inject field io.quarkus.it.spring.data.jpa.PersonRepository_bcb609283abd593bd6e8ce51955e38a20ba9d528Impl#customImplClass1,
- @Inject field io.quarkus.it.spring.data.jpa.PersonRepository_bcb609283abd593bd6e8ce51955e38a20ba9d528Impl#customImplClass2

jayee committed:
MongoDB-Client: Make it possible to override providers in the default codec registry

jayee committed:
Refactored the creation of the registry

aoudiamoncef committed:
doc(*): Clean loggers

fix writing extensions format

stuartwdouglas committed:
Specify keycloak version in tests

stuartwdouglas committed:
Add the notion of a current vert.x request

This allows the Vert.x routing context to be injected

stuartwdouglas committed:
Remove Undertow dependency from Opentracing

Fixes #5419

mkouba committed:
Scheduler - do not use interceptor to activate request context per exec

- otherwise the scheduled bean must be proxyable

jmartisk committed:
Change compile quarkus:dev to just quarkus:dev in Maven plugin's output

gsmet committed:
Add the Jackson exception to the Writing native applications tips guide

Follow-up of
https://stackoverflow.com/questions/58911471/strange-lombok-behaviour-with-quarkus-and-jackson/58911908

mkouba committed:
Document lifecycle of JAX-RS resources

- resolves #1371

gsmet committed:
Add some standards to the README

Fix #2404

machi1990 committed:
docs: mention the quarkus.scheduler.cron-type property

Follows up https://github.com/quarkusio/quarkus/pull/5481

geoand committed:
Place json configuration files next to thin jar

This makes it very easy and predicable to configure
add extra json files to configure GraalVM natives.
Furthermore, this makes these fails available to
docker when a container build of the native image
is being used.

Fixes: #5390

gsmet committed:
Also consider XmlType as triggering the JAXB inclusion

XmlRootElement is optional so we need to trigger the JAXB inclusion with
XmlType too.

gsmet committed:
Remove the substitution for LocatableAnnotation and register proxies

This implementation is important and used by JAXB. We just need to
register proxies for the combinations of JAXB annotations and Locatable.

gsmet committed:
Register W3CDomHandler if we have a @XmlAnyElement annotation

kenfinnigan committed:
Update to SmallRye OpenAPI 1.1.20

stuartwdouglas committed:
Terminate request scope before sending JAX-RS response

This gives more consistent span ordering for Opentracing

mkouba committed:
Docs Reactive Routes - minor updates/fixes

- use RoutingExchange
- @Route is repeatable

FroMage committed:
Document replacement for HttpServletRequest

ia3andy commented:
Moved to https://github.com/quarkusio/quarkus/pull/5585
